### ISSUE TITLE
Unexpected error dialog after cancel (security) information dialog

### ISSUE THREAD

**Issue by barthel** [karakun/OpenWebStart#80]
OpenWebStart 0.5.1
macOS 10.15.1

The application starts and this (security) information dialog is shown:
![Screenshot 2019-11-14 at 10 49 55](https://user-images.githubusercontent.com/1170328/68846349-3bd37e80-06cd-11ea-8a70-30027ae7dfb6.png)

After cancel this dialog a new unexpected error dialog is shown:
![Screenshot 2019-11-14 at 10 50 06](https://user-images.githubusercontent.com/1170328/68845999-9c15f080-06cc-11ea-9622-89cea2d68024.png)

```
[...]
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [ERROR  ] net.sourceforge.jnlp.Launcher: ERROR net.sourceforge.jnlp.LaunchException: Fatal: Initialization Error: Could not initialize application. The application has not been initialized, for more information execute javaws from the command line.     at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:669)     at net.sourceforge.jnlp.Launcher.launchApplication(Launcher.java:411)     at net.sourceforge.jnlp.Launcher.access$300(Launcher.java:69)     at net.sourceforge.jnlp.Launcher$TgThread.run(Launcher.java:756) Caused by: net.sourceforge.jnlp.LaunchException: The application uses non-codebase resources, which do match its Application-Library-Allowable-Codebase Attribute, but was blocked from running by the user.     at net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker.checkApplicationLibraryAllowableCodebaseAttribute(ManifestAttributesChecker.java:423)     at net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker.checkAll(ManifestAttributesChecker.java:130)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.initializeResources(JNLPClassLoader.java:829)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.<init>(JNLPClassLoader.java:347)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.createInstance(JNLPClassLoader.java:420)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:492)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:465)     at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:661)     ... 3 more 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [ERROR  ] net.sourceforge.jnlp.AbstractLaunchHandler: netx: Initialization Error: Could not initialize application. (The application uses non-codebase resources, which do match its Application-Library-Allowable-Codebase Attribute, but was blocked from running by the user.) net.sourceforge.jnlp.LaunchException: Fatal: Initialization Error: Could not initialize application. The application has not been initialized, for more information execute javaws from the command line.     at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:669)     at net.sourceforge.jnlp.Launcher.launchApplication(Launcher.java:411)     at net.sourceforge.jnlp.Launcher.access$300(Launcher.java:69)     at net.sourceforge.jnlp.Launcher$TgThread.run(Launcher.java:756) Caused by: net.sourceforge.jnlp.LaunchException: The application uses non-codebase resources, which do match its Application-Library-Allowable-Codebase Attribute, but was blocked from running by the user.     at net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker.checkApplicationLibraryAllowableCodebaseAttribute(ManifestAttributesChecker.java:423)     at net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker.checkAll(ManifestAttributesChecker.java:130)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.initializeResources(JNLPClassLoader.java:829)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.<init>(JNLPClassLoader.java:347)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.createInstance(JNLPClassLoader.java:420)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:492)     at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:465)     at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:661)     ... 3 more 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.client.parts.dialogs.security.SecurityDialogs: Decided action for matching alaca at https://my.application.codebase.url/ was NO 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker: The resources from https://my.application.codebase.url do  match the location in Application-Library-Allowable-Codebase Attribute * . Continuing. 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker: The resources from file:/Users/XXXXXXXXXX/Downloads do  match the location in Application-Library-Allowable-Codebase Attribute * . Continuing. 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [WARNING] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker: The resource 'file:/Users/XXXXXXXXXX/Downloads/my.application.jnlp' is not from codebase 'https://my.application.codebase.url/' 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker: OK - 'https://my.application.codebase.url' is from codebase 'https://my.application.codebase.url/'. 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [WARNING] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker: Warning! 'file:/Users/XXXXXXXXXX/Downloads' is NOT from codebase 'https://my.application.codebase.url/'. 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [INFO  ] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker: Codebase matches codebase manifest attribute, and application is signed. Continuing. See: http://docs.oracle.com/javase/7/docs/technotes/guides/jweb/security/no_redeploy.html for details. 
[ITW-JAVAWS-CLIENT][MESSAGE_ALL][Thu Nov 14 10:58:01 CET 2019][sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)] NETX Thread# 29c1edb2, name My Application Installation: Thu Nov 14 10:58:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesChecker: Trusted Only manifest attribute not found. Continuing. 
[...]
```

If this dialog was cancelled once, this application can't be start anymore. The error dialog will shown at every start.

**Comment by hendrikebbers** [comment]
The initial dialog is showed because the JNLP does not contain a HTTP-URL to the origin of the JNLP. Therefore WebStart will use the local JNLP in cache instead of the "online" JNLP version. So the easiest way would be to change the JNLP file.

We added a flag to the master that allows to always use files from the filesystem. This will be part of 1.0 that is targeted for next week. This is currently not available in the settings ui and must be specified by hand in the config file:

/user-home/.config/icedtea-web/deployment.properties

You can add the following property:
deployment.assumeFileSystemInCodebase=true

You can find more information here:
https://openwebstart.com/configuration/

Once this is done you should not come anymore in such situation. But next to this we have the issue that OpenWebStart somewhere stores you decision. At the moment I can not say why this happens and where it is stored without having a deeper look in the code.

**Comment by barthel** [comment]
> allows to always use files from the filesystem

Will this be the default behaviour?

**Comment by hendrikebbers** [comment]
> Will this be the default behaviour?

Until now it was not planed to be default behaviour in 1.0 since we are not 100% sure about side effects. Adding @sclassen for discussion about making it the default ;)

**Comment by barthel** [comment]
This behavior occurs when a browser (e.g. Internet Explorer) store the JNLP file into a temp. directory to start the OpenWebStart process.

**Comment by hendrikebbers** [comment]
https://github.com/AdoptOpenJDK/IcedTea-Web/pull/510 will add the properties to the security tap in the setting. At least it can be changed in the UI if it won't be default activated in 1.0

**Comment by sclassen** [comment]
The second dialog will be addressed in #79
For the first dialog we added a flag to allow resources from the local file system without showing a dialog.
This flag can also be set using the unattended installation ([see configuration](https://openwebstart.com/configuration/))
